### PR TITLE
fix: remove broken build request hack

### PR DESCRIPTION
**Pull request by ronag**
## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by metcoder95 — APPROVED** [review]

**Comment by mcollina** [comment]
tests are failing

File: lib/core/util.js [hunk 1]
```diff
@@ -516,6 +516,7 @@ kEnumerableProperty.enumerable = true
 module.exports = {
   kEnumerableProperty,
   nop,
+
```
Comment (ronag) [inline lib/core/util.js:519]:
```suggestion
```

**Comment by jessezhang91** [comment]
Hey @ronag, I'm tracing through [this issue](https://github.com/getsentry/sentry-javascript/issues/10936) that seems to involve the `undici:request:create` diagnostics channel having changed in schema for the headers due to this PR (was string, now array). I just wanted to confirm that's the right type given the [documentation here](https://github.com/nodejs/undici/blob/main/docs/docs/api/DiagnosticsChannel.md) is likely incorrect now as well. I'm happy to make a PR to update the documentation as well.

**Comment by ronag** [comment]
It's correct. PR would be appreciated.

**Pull request merged**